internal/controller/eventing/controller.go

```diff
@@ -214,9 +214,6 @@ func (r *Reconciler) reconcileNATSBackend(ctx context.Context, eventing *eventin
 
 	// set NATSAvailable condition to true and update status
 	eventing.Status.SetNATSAvailableConditionToTrue()
-	if err := r.syncEventingStatus(ctx, eventing, log); err != nil {
-		return ctrl.Result{}, err
-	}
 
```

Comment:
What you deleted reflects the NATS availability status immediately. Let's says the condition is not stored in ETCD and EPP deployment is still in-progress, then the condition still displays NATS unavailable (although available).
Hence, I would leave the logic.